### ISSUE TITLE
TensorRT inference issue : tensor output is not properly generating the tensor output using p5 scaled yolo v4

### ISSUE THREAD

**Issue by BloomBerry** [WongKinYiu/ScaledYOLOv4#257]
Hi folks,

I am having TensorRT inference issue using p5 scaled Yolov4.

The tensorRT inference code I was using is based on https://github.com/spmallick/learnopencv/blob/master/PyTorch-ONNX-TensorRT/trt_inference.py

The warning i'm facing is 
Scanning labels /data/active_labtest_text/valid.cache (36242 found, 0 missing, 0 empty, 0 duplicate, for 36242 images): 100%|██████████| 36242/36242 [00:01<00:00, 19497.94it/s]
               Class      Images     Targets           P           R      mAP@.5  mAP@.5:.95:   0%|          | 0/36242 [00:00<?, ?it/s][TensorRT] ERROR: [hardwareContext.cpp::configure::92] Error Code 1: Cudnn (CUDNN_STATUS_MAPPING_ERROR)
               Class      Images     Targets           P           R      mAP@.5  mAP@.5:.95:   0%|          | 1/36242 [00:00<3:13:49,  3.12it/s][TensorRT] ERROR: [hardwareContext.cpp::configure::92] Error Code 1: Cudnn (CUDNN_STATUS_MAPPING_ERROR)
[TensorRT] ERROR: [hardwareContext.cpp::configure::92] Error Code 1: Cudnn (CUDNN_STATUS_MAPPING_ERROR)
               Class

The inference code I'm using is as follows:

```
h_out = execute_async(context, engine, builder, img_cpu)

def execute_async(context, engine, builder, tensor_inputs):
    inputs, outputs, bindings, stream = [], [], [], []
    # get sizes of in / out and allocate memory required for in / out data
    for binding in engine:
        size = trt.volume(engine.get_binding_shape(binding)) * builder.max_batch_size
        dtype = trt.nptype(engine.get_binding_dtype(binding))
        host_mem = cuda.pagelocked_empty(size, dtype)
        device_mem = cuda.mem_alloc(host_mem.nbytes)
        bindings.append(int(device_mem))
        if engine.binding_is_input(binding):
            inputs.append(HostDeviceMem(host_mem, device_mem))
        else:
            outputs.append(HostDeviceMem(host_mem, device_mem))

    # Create a stream in which to cpy inputs/outputs and run inference.
    stream = cuda.Stream()

    for input, tensor_input in zip(inputs, tensor_inputs):
        numpy_input = np.array(tensor_input.numpy(), dtype=np.float32, order='C').ravel()
        np.copyto(input.host, numpy_input)

    [cuda.memcpy_htod_async(inp.device, inp.host, stream) for inp in inputs]

    context.execute_async_v2(bindings=bindings, stream_handle=stream.handle)

    [cuda.memcpy_dtoh_async(out.host, out.device, stream) for out in outputs]

    stream.synchronize()

    return [out.host for out in outputs]
```

where context, engine, builder is generated with following code

```
def trtConversion(self, input_path, output_path, weights=None, data=None, batch_size=None, img_size=None,
                   conf_thres=None, iou_thres=None, save_txt=None, save_json=None):
        weight_name = weights.split('/')[-1].split('.')[0]
        onnx_path = output_path + '/' + weight_name + '.onnx'
        engine_path = output_path + '/' + weight_name + '.engine'
        img = torch.zeros((1, 3, img_size, img_size))

        # Load model
        model = attempt_load(weights, map_location=torch.device('cpu'))  # load FP32 model
        # Update model
        for k, m in model.named_modules():
            m._non_persistent_buffers_set = set()  # pytorch 1.6.0 compatability
            if hasattr(m, 'act') and isinstance(m.act, models.common.Mish):
                m.act = Mish()
            if isinstance(m, models.common.Mish):
                m = Mish()
            if isinstance(m, models.common.BottleneckCSP) or isinstance(m, models.common.BottleneckCSP2) \
                    or isinstance(m, models.common.SPPCSP):
                if isinstance(m.bn, nn.SyncBatchNorm):
                    bn = nn.BatchNorm2d(m.bn.num_features, eps=m.bn.eps, momentum=m.bn.momentum)
                    bn.training = False
                    bn._buffers = m.bn._buffers
                    bn._non_persistent_buffers_set = set()
                    m.bn = bn
                if isinstance(m.act, models.common.Mish):
                    m.act = Mish()  # assign activation
            # if isinstance(m, models.yolo.Detect):
            #     m.forward = m.forward_export  # assign forward (optional)

        model.eval()
        model.model[-1].export = True  # set Detect() layer export=True
        def build_engine(onnx_file_path, engine_path):
            # convert the pytorch model to onnx
            TRT_LOGGER = trt.Logger()
            EXPLICIT_BATCH = 1 << (int)(trt.NetworkDefinitionCreationFlag.EXPLICIT_BATCH)
            # initialize TensorRT engine and parse ONNX model
            with trt.Builder(TRT_LOGGER) as builder, builder.create_network(EXPLICIT_BATCH) as network, \
                    builder.create_builder_config() as config, trt.OnnxParser(network, TRT_LOGGER) as parser:
                # parse ONNX
                with open(onnx_file_path, 'rb') as model:
                    print("Beginning ONNX file parsing")
                    parser.parse(model.read())
                    print("Completed parsing of ONNX file")

                # allow TensorRT to use up to 1GB of GPU memory for tactic selection
                config.max_workspace_size = 1 << 30  # 1 [GB]
                # we have only one image in batch
                builder.max_batch_size = 1
                # generate TensorRT engine optimized for the target platform
                print("Building an engine...")
                engine = builder.build_engine(network, config)

                context = engine.create_execution_context()
                print("Complted creating Engine")

                return engine, context, builder
        if not os.path.exists(onnx_path):
            torch.onnx.export(model, img, onnx_path, input_names=['input'],
                          output_names=['output'], opset_version=11)

        # initialize TensorRT engine and parse ONNX model
        engine, context, builder = build_engine(onnx_path, engine_path)
        return engine, context, builder
```

Thanks in advance
